sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -219,19 +219,18 @@ protected PartitionLoadBalancer()
             // Find an ownership to claim and try to claim it.  The method will return null if this instance was not eligible to
             // increase its ownership list, if no claimable ownership could be found or if a claim attempt has failed.
 
-            var claimedOwnership = await FindAndClaimOwnershipAsync(completeOwnershipList, unclaimedPartitions, partitionIds.Length, cancellationToken).ConfigureAwait(false);
+            var (claimAttempted, claimedOwnership) = await FindAndClaimOwnershipAsync(completeOwnershipList, unclaimedPartitions, partitionIds.Length, cancellationToken).ConfigureAwait(false);
 
             if (claimedOwnership != null)
             {
                 InstanceOwnership[claimedOwnership.PartitionId] = claimedOwnership;
-                unclaimedPartitions.Remove(claimedOwnership.PartitionId);
             }
 
-            // Update the balanced state.  Consider the load balanced if this processor has its minimum share of partitions, no partitions
-            // remain unclaimed and this cycle did not claim a partition.
+            // Update the balanced state.  Consider the load balanced if this processor has its minimum share of partitions and did not
+            // attempt to claim a partition.
 
             var minimumDesiredPartitions = partitionIds.Length / ActiveOwnershipWithDistribution.Keys.Count;
-            IsBalanced = ((InstanceOwnership.Count >= minimumDesiredPartitions) && (unclaimedPartitions.Count <= 0) && (claimedOwnership == null));
+            IsBalanced = ((InstanceOwnership.Count >= minimumDesiredPartitions) && (!claimAttempted));
```

Comment:
Do we need a test to validate this change in `IsBalanced` behavior?

Reply:
I'm comfortable with the rigor in the existing test scenarios.

Ideally, we'd cover the additional case of  "I have my fair share of partitions but tried to claim anyway and failed" case.  That's really an artificial case, as it would require overriding the algorithm that determines if a claim is needed.  While there would be some value there, the implementation is a couple of nested levels of private.   I don't believe the benefit is worth the cost in this case.   I'm going to punt on this for now but we can revisit if we see a reason to do so in the future.